### PR TITLE
Feat: Import functions that are written as ES Modules

### PR DESCRIPTION
**Pull request by lholmquist**
Currently only the common js format can be used to create functions, something like:

```
module.exports = function handler(context) { ... }
```
and then to use this function: `faas-js-runtime index.js`

This adds the ability to create the function in the ES module format.  Function could look something like this:

```
const handle = async function(context) => { ... };

// Export the function
export { handle };
```

and then to use it `faas-js-runtime index.mjs`

You can also use this `faas-js-runtime index.js` if the functions package.json has the `type=module` field

### CODE REVIEW

File: test/test-function-loading.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+const test = require('tape');
+const path = require('path');
+const { loadFunction } = require('../lib/function-loader.js');
+
+const fixtureDir = path.join(__dirname, 'fixtures');
+
+test('Loads a CJS function', async t => {
+  await loadFunction(path.join(fixtureDir, 'cjs-module', 'index.js'));
```
Comment (lance) [inline test/test-function-loading.js:8]: Can you add a check here and in the other two tests similar to this?
```suggestion
  const fn = await loadFunction(path.join(fixtureDir, 'cjs-module', 'index.js'));
  t.equal(typeof fn, 'function');
```

File: test/fixtures/esm-module-type-module/package-lock.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (lance) [inline test/fixtures/esm-module-type-module/package-lock.json:1]: This file isn't really needed for test fixtures

File: test/fixtures/esm-module-mjs/package-lock.json [hunk 3]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (lance) [inline test/fixtures/esm-module-mjs/package-lock.json:1]: This file isn't really needed for test fixtures

File: test/fixtures/cjs-module/package-lock.json [hunk 4]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (lance) [inline test/fixtures/cjs-module/package-lock.json:1]: This file isn't really needed for test fixtures

**Review by lance — COMMENTED** [review]
This is great @lholmquist thanks! Just a couple of small suggestions.

File: test/test-function-loading.js (hunk 1 above)
Comment (lholmquist) [reply test/test-function-loading.js:8]: will do

**Review by lance — APPROVED** [review]
/lgtm
Thanks!

**Pull request merged**